UnityProject/Assets/Scripts/UI/Cargo/GUI_SecurityRecordsEntryPage.cs

```diff
@@ -0,0 +1,223 @@
+	private SecurityRecordCrime currentlyEditingCrime;
+	[SerializeField]
+	private NetCompositeImage photoFront;
+
+	public void OnOpen(SecurityRecord recordToOpen, GUI_SecurityRecords recordsTab)
+	{
+		record = recordToOpen;
+		securityRecordsTab = recordsTab;
+		UpdateEntry();
+	}
+
+	private void OnEnable()
+	{
+		ClosePopup();
+	}
+
+	public void IdNameUpdate()
+	{
+		if (securityRecordsTab == null)
+			return;
+
+		IDCard id = securityRecordsTab.InsertedCard;
+		string str;
+
+		if (id != null)
+			str = $"{id.RegisteredName}, {id.GetJobType.ToString()}";
+		else
+			str = "********";
+		idNameText.SetValue = str;
+	}
+
+	public void RemoveID()
+	{
+		securityRecordsTab.RemoveId();
+		IdNameUpdate();
+	}
+
+	private void UpdateEntry()
+	{
+		if (!CustomNetworkManager.Instance._isServer)
+			return;
+		if (record == null)
+			return;
+		nameText.SetValue = record.EntryName;
+		idText.SetValue = record.ID;
+		sexText.SetValue = record.Sex;
+		ageText.SetValue = record.Age;
+		speciesText.SetValue = record.Species;
+		rankText.SetValue = record.Rank;
+		fingerprintText.SetValue = record.Fingerprints;
+		statusButtonText.SetValue = record.Status.ToString();
+		photoFront.SetValue = "";
+		if (record.player != null)
+		{
+			photoFront.SetValue = record.player.netId.ToString();
+		}
+
+		IdNameUpdate();
+		UpdateCrimesList();
+	}
+
+	public void ChangeStatus()
+	{
+		switch (record.Status)
+		{
+			case SecurityStatus.None:
+				record.Status = SecurityStatus.Arrest;
+				break;
+			case SecurityStatus.Arrest:
+				record.Status = SecurityStatus.Parole;
+				break;
+			case SecurityStatus.Parole:
+				record.Status = SecurityStatus.None;
+				break;
+		}
+		statusButtonText.SetValue = record.Status.ToString();
+	}
+
+	/// <summary>
+	/// Opens popup locally. Whole interaction cycle look like this:
+	/// 1. Client opens Popup and sets currenty edited field on the server.
+	/// 2. Client confirms edit in popup, popup closes locally.
+	/// 3. Server sets fields with values from popup.
+	/// </summary>
+	public void OpenPopup()
+	{
+		popupWindow.SetActive(true);
+	}
+
+	/// <summary>
+	/// Set field to edit in popup.
+	/// Used for info entry (name, age, etc.)
+	/// </summary>
+	public void SetEditingField(NetLabel fieldToEdit)
+	{
+		currentlyEditingField = fieldToEdit;
+	}
+
+	/// <summary>
+	/// Set Editing field for crime entry.
+	/// </summary>
+	public void SetEditingField(NetLabel fieldToEdit, SecurityRecordCrime crimeToEdit)
+	{
+		currentlyEditingField = fieldToEdit;
+		currentlyEditingCrime = crimeToEdit;
+	}
+
+	/// <summary>
+	/// Sets currentlyEditingField value to sent value.
+	/// The way it is done is bad, I just couldn't come up with better one.
+	/// </summary>
+	/// <param name="value">String to set in field.</param>
+	public void ConfirmPopup(string value)
+	{
+		currentlyEditingField.SetValue = value;
+		string nameBeforeIndex = currentlyEditingField.name.Split('~')[0];
+		switch (nameBeforeIndex)
```

Comment:
swap this to an int or enum, strings are really slow

Reply:
Not sure how to do it here - this part can get a lot of different NetLabels at input, and the only thing that distinguishes it from others is its name.